client/test-utils/flushPromises.ts

```diff
@@ -0,0 +1,9 @@
+function flushPromises() {
```

Comment:
This is almost identical code to https://github.com/kentor/flush-promises/blob/master/index.js so wondering as this'll be a dev dependency whether we just use the npm `flush-promises` package instead, thoughts?

Reply:
we are using flush-promises in the event-bus for tests so maybe a reason for standardisation albeit quite weak if we are moving to many repos.

Reply:
I'm for moving to use the published package then. Can't see many benefits of having to maintain the dozen lines of code ourself. Also we're not covering test-utils with unit tests where as the package does have coverage.

Reply:
scrap that, this is evaluating to setInterval which was causing the console errors. Lets stick with the local code instead.
 https://github.com/kentor/flush-promises/blob/46f58770b14fb74ce1ff27da00837c7e722b9d06/index.js#L1

Reply:
Also check here: https://medium.com/better-programming/process-nexttick-vs-setimmediate-vs-settimeout-explained-wrt-different-event-loop-phases-c0506b12921d